gui/src/pages/NewProjectConfig.svelte

```diff
@@ -0,0 +1,149 @@
+<script lang="ts">
+  import Layout from '../components/Layout.svelte';
+  import Textbox from '../components/Textbox.svelte';
+  import Spinner from '../components/Spinner.svelte';
+  import ErrorLabel from '../components/ErrorLabel.svelte';
+  import SubmitButton from '../components/form/SubmitButton.svelte';
+  import CenterFormPanel from '../components/form/CenterFormPanel.svelte';
+  import { api, isClientError } from '../lib/api';
+  import * as router from 'svelte-spa-router';
+
+  export let params = { starter: '' };
+
+  const { starter } = params;
+  const isEmptyProject = starter === 'empty'; // XXX Use this with the `createProject` api to avoid a second (template URL) parameter if the project is empty.
+
+  let name = starter;
+  let error: Error | null = null;
+
+  let workingDirectory: string | null = null;
+
+  const setWorkingDirectory = (dir: string) => {
+    workingDirectory = dir;
+  };
+
+  (async () => {
```

Comment:
this is not a great pattern, since it introduces a potential race condition. Each time this code runs, it is racing with event handlers that set the working directory. Instead, consider changing the working directory to be of type Promise<string> and initializing it immediately that way.

Reply:
Okay, I believe this is fixed now, thanks @BenElgar